packages/core/types/src/types/core/index.ts

```diff
@@ -1,9 +1,20 @@
+/**
+ * Note: linting will erroneously raise errors if a namespaced type has the same name as on exported by Strapi
+ * To work around it, use a pattern such as:
+ *
+ * import * as ConfigImport from './config'
+ * export type Config = typeof ConfigImport
+ *
+ * which causes linting to see the namespaces correctly.
+ */
+import * as ConfigImport from './config';
+
+export type Config = typeof ConfigImport;
```

Comment:
Yeah. In the parent file, there is an error that Server is a duplicate export. We tried commenting out the actual Server in there and confirmed that our Config Server is not actually exported, so it appears to be a purely linting issue. So I don't like this hack (changing valid code to something less readable is the opposite of what lint should be doing) but it's the best we could come up with.

Reply:
Just a note, I noticed last week that this hack might not even work. So I'll have to keep trying some other things to solve the error.

I also wonder if renaming everything *Input, like ServerInput, AdminInput, etc, would be a solution since we use the Input suffix in other areas like RouterInput. What do you think about that?

Reply:
That's weird...
If it's only a lint issue there might be some configuration work to do.
Regarding the naming, you are referring to the config, right? If nothing else works I guess we can adapt and go for it, but let's try to keep the names as simple & straight-forward as possible

Reply:
Yeah, it seems to be only a lint error as far as I can tell. People have been reporting it since 2020 and it's still unresolved. I think we can solve it without renaming, but the renaming may be preferable anyway, and if that's the case the duplicate names can be a problem for the future 😆